### ISSUE TITLE
RTCRtpTransceiverInit: receive parameters

### ISSUE THREAD

**Issue by mparis** [w3c/webrtc-pc#622]
Hello,
I see that a transceiver can be configured with parameters for sending [1].

I think that having parameters for receiving is also interesting. Wiht these parameters we can configure for example:

- Max recv bandwidth (exposed in the SDP as "b=AS:X")
- Congestion Control options
- JitterBuffer configuration like minSize, maxSize, etc.
- etc.

There are some comments in discuss-webrtc asking for a way of configure this kind of things.
What do you think about this idea?

Thanks!!

Refs
[1] http://w3c.github.io/webrtc-pc/#widl-RTCRtpTransceiverInit-sendEncodings

**Comment by aboba** [comment]
RTCTransceiverInit dictionary includes sendEncodings because the RTCRtpEncodingParameter dictionary is largely for use by RtpSender objects, not RtpReceiver objects.  Based on what is in Section 5.2.4, here is a table of which attributes are relevant to senders and receivers: 


Attribute	                                          Type	                      Receiver/Sender              Read/Write
-------------------------                 -------------------------                -------------------------            ------------------------
ssrc	                                       unsigned long	                      Sender                             Read-only
fec	                                       RTCRtpFecParameters	      Receiver/Sender              Read-only
rtx	                                       RTCRtpRtxParameters	      Receiver/Sender              Read-only
active	                               boolean	                              Sender                             Read/Write
priority	                               RTCPriorityType	              Sender                             Read/Write
maxBitrate	                       unsigned long	                      Sender                             Read/Write
maxFramerate	                       unsigned long	                      Sender                             Read/Write
scaleResolutionDownBy	       double	                              Sender                             Read/Write
rid                                           DOMString	                      Sender                             Read/Write

**Comment by alvestrand** [comment]
The transceiver doesn't seem to be the right surface to control receiver options.
If there are specific controls that need exposing for a particular usage, please propose them as separate issues, and we'll discuss what the most appropriate placement is.

**Comment by mparis** [comment]
@alvestrand, if it not this is not the right place, why does RTCRtpTransceiverInit have sendEncodings params?
To keep the API symetric, send and recv params should be in the same place, don't they?

**Comment by pthatcherg** [comment]
sendEncodings are a parameter meant for the RtpSender.  They are passed into addTransceiver because it's impossible to construct an RtpSender without a full RtpTransceiver, and it's impossible to add encodings after creating the RtpSender (with the 1.0 API;  hopefully future versions of the API will not have this restriction).

The RtpReceiver could potentially have controls/parameters for bandwidth, Congestion Control, JitterBuffer, etc.  But I think that's outside the scope of 1.0.

**Comment by mparis** [comment]
@pthatcherg why is it outside the scope of 1.0?
Until now, at least the max recv bandwidth can be changed munging the "b=AS x" entry in the SDP.

**Comment by pthatcherg** [comment]
They are outside the scope of 1.0 because many months ago (coming up on a year?), we said, basically, "this is what will be in 1.0, and that's it; everything else will have to be a later version".  We have to finish 1.0 at some point, so we can't just keep adding more things, at least not until we finish 1.0.

**Comment by mparis** [comment]
OK, I agree with this philosophy, if not we will never have anything.
So can be keep this issue for the next version?

**Comment by aboba** [comment]
I don't think that this is just an issue of whether RTCRtpReceiver.setParameters() is out of scope for WebRTC 1.0.  In SDP, bandwidth limits such as RFC 3890 express a maximum bandwidth for the session.  This limit could be provided as a sender setting as it is in WebRTC 1.0, but having it be a receiver setting does not make all that much sense.  Once the sender has decided to send a certain bandwidth to the receiver, what can the receiver do about it?  Drop packets?  Send REMB or other congestion control messages back to the sender indicating a maximum available bandwidth?  It seems to me that this is not just about adding a knob, but also presenting a clear statement about what the desired functionality that knob would enable and why it makes sense. 

After all, ORTC has the ability to set receiver parameters (via the receive(parameters) call) but the spec currently doesn't envisage a useful role for maxBitrate on the receiver.

**Comment by alvestrand** [comment]
Looking at the original post:
Jitter buffer parameters are receiver parameters if they exist at all, not transceiver parameters.
Congestion control options are hard to control because we don't know what options we want yet.
Setting b=as; values in SDP is probably a transceiver parameter, because transceivers are the right place for controls that affect SDP.

But as pthatcherg said, expanding the scope of 1.0 is out of scope for now. I'm attaching the LATER label.
(when the time comes up to review LATER-labelled items, I'm likely to ask to split this into its component requests - they're vastly different.)
